### PR TITLE
Reduce the size of the index buffers.

### PR DESCRIPTION
**Pull request by robertoraggi**
Prefer to use uint16 buffers when possible.

Signed-off-by: Roberto Raggi <roberto.raggi@here.com>

### CODE REVIEW

File: @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts [hunk 1]
```diff
@@ -137,6 +137,38 @@ const MAX_CORNER_ANGLE = Math.PI / 8;
...
+
+function createIndexBuffer(elements: ArrayLike<number>) {
+    return maxValue(elements) > 65535 ? new Uint32Array(elements) : new Uint16Array(elements);
+}
+
+function createIndexBufferAttribute(
+    elements: ArrayLike<number>,
+    name: string = "index"
+): BufferAttribute {
+    const type = maxValue(elements) > 65535 ? "uint32" : "uint16";
```
Comment (nzjony) [inline @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts:161]: could you pass in an optional value which has the vertices.length (if available), if it is less than 65k, then you can avoid the maxValue function call?

File: @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts [hunk 2]
```diff
@@ -137,6 +137,38 @@ const MAX_CORNER_ANGLE = Math.PI / 8;
  */
 const INVALID_ARRAY_INDEX = -1;
 
+function maxValue(array: ArrayLike<number>): number {
```
Comment (MPursche) [inline @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts:140]: Given that the index buffer does not contain indices that are out of range: the highest index can also be determined with `posAttr.count` which should be a bit faster than looping over the index buffer.

File: @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts (hunk 1 above)
Comment (ninok) [inline @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts:161]: In native we even used 16 bit when exceeding 65K and did multiple draw calls with [glDrawElementsBaseVertex.](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glDrawElementsBaseVertex.xhtml).
Not sure if sth. equivalent exists in WebGL.

Comment (robertoraggi) [reply @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts:161]: it's not vertexLength, it's the max value of the array.

Comment (MPursche) [reply @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts:161]: But shouldn't the highest value of the index buffer equal(or be smaller than) `posAttr.count - 1`?

Comment (nzjony) [reply @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts:161]: why isn't it vertextLength? (well actually vertextLength-1)

File: @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts (hunk 2 above)
Comment (robertoraggi) [reply @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts:140]: ok

File: @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts (hunk 1 above)
Comment (MPursche) [reply @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts:161]: I think there is only this extension for WebGL 2: https://www.khronos.org/registry/webgl/extensions/WEBGL_draw_instanced_base_vertex_base_instance/. So not very likely that this is supported by many devices.

Comment (robertoraggi) [reply @here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts:161]: sure, using vertex.length may be OK with the way we create geometries right now.

**Comment by robertoraggi** [comment]
Retests this please

**Pull request merged**